pkgs/development/tools/misc/saleae-logic-2/default.nix

```diff
@@ -1,11 +1,22 @@
-{ lib, fetchurl, appimageTools, gtk3 }:
+{ lib, fetchurl, makeDesktopItem, appimageTools, gtk3 }:
 let
   name = "saleae-logic-2";
   version = "2.3.37";
   src = fetchurl {
     url = "https://downloads.saleae.com/logic2/Logic-${version}-master.AppImage";
     sha256 = "0jclzd4s1r6h2p1r0vhmzz3jnwpp7d41g70lcamrsxidxrmm8d45";
   };
+  desktopItem = makeDesktopItem {
+    name = name;
+    exec = name;
+    icon = appimageTools.extractType2 {
+      inherit name src;
+    } + "/usr/share/icons/hicolor/256x256/apps/Logic.png";
```

Comment:
instead of this move the icon to `$out/share/pixmaps/Logic.png` in  `extraInstallCommands` and set the icon to `Logic`